openhands/core/config/extended_config.py

```diff
@@ -0,0 +1,28 @@
+class ExtendedConfig:
+    """Configuration for extended functionalities.
+
+    Attributes:
+        Values depend on the defined section
+    """
+
+    extended: dict = {}
+
+    def __str__(self):
+        attr_str = []
+        for key_name, dict_value in self.extended.items():
+            for attr_name, attr_value in dict_value.items():
+                attr_str.append(f'[{key_name}]{attr_name}={repr(attr_value)}')
+        return f"ExtendedConfig({', '.join(attr_str)})"
+
+    def add_dict(self, key: str, extended_config_dict: dict) -> 'ExtendedConfig':
+        self.extended[key] = extended_config_dict
```

Comment:
In that case, we don't need a key in extended to be a dict, I would suggest to keep it simple, just hold its dict of primitives I think

Reply:
oh, I see so  have a unique extended section ? instead of allowing other sections to be added by other tool providers ? and they would add their value in that 'extended' ?

Reply:
Yes, I'd suggest to deal with it when we come to it.